code/modules/food_and_drinks/kitchen_machinery/kitchen_machine.dm

```diff
@@ -156,7 +156,54 @@
 		//G.reagents.trans_to(src,G.amount_per_transfer_from_this)
 	else if(istype(O,/obj/item/weapon/grab))
 		var/obj/item/weapon/grab/G = O
-		to_chat(user, "<span class='alert'>This is ridiculous. You can not fit \the [G.affecting] in this [src].</span>")
+		if(!istype(src, /obj/machinery/kitchen_machine/oven) && !istype(src, /obj/machinery/kitchen_machine/grill))
+			to_chat(user, "<span class='alert'>This is ridiculous. You can not fit \the [G.affecting] in this [src].</span>")
+			return 1
+		if(!iscarbon(G.affecting))
+			to_chat(user, "<span class='warning'>You can only harm carbon-based creatures this way!</span>")
+			return 1
+//		if(G.affecting.stat == DEAD)
+//			to_chat(user, "<span class='warning'>[G.affecting] is dead!</span>")
+//			return 1
+		if(G.state < GRAB_AGGRESSIVE)
+			to_chat(user, "<span class='warning'>You need a better grip to do that!</span>")
+			return 1
+		var/mob/living/carbon/human/C = G.affecting
+		var/obj/item/organ/external/head/head = C.get_organ("head")
```

Comment:
Same here. If the line aboe this one doesn't assign anything to C, because it's a carbon, but not a human, this proc will get called on nothing.